packages/auth-providers/supabase/middleware/src/index.ts

```diff
@@ -68,6 +70,8 @@ const createSupabaseAuthMiddleware = ({
         isAuthenticated: !!currentUser,
         hasError: false,
         userMetadata: userMetadata || currentUser,
+        cookieHeader,
+        roles: extractRoles ? extractRoles(decoded) : [],
```

Comment:
I am trying to think of a case where the decoded data from the authDecoder is needed to get roles -- and not just the currentUser.

I guess one doesn't have to set roles inside `getCurrentUser` but typically for the auth providers we do.

Though this does give some flexibility to define roles outside the currentUser .... which might make ABAC permission easier in the future. Ie - check a permission file/config or lookup db --- for if the username starts with admin , then then get the admin role.

Think will have to use it a bit and get the feel for it, but makes sense and can see ow it will work.

Reply:
I think for example, if you add roles in supabase user_metadata, it doesn't come through as currentUser.roles!